beacon-chain/rpc/eth/light-client/handlers.go

```diff
@@ -0,0 +1,402 @@
+package lightclient
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"net/http"
+	"strconv"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/gorilla/mux"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/rpc/eth/shared"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/v4/config/params"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/interfaces"
+	types "github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
+	http2 "github.com/prysmaticlabs/prysm/v4/network/http"
+	ethpbv2 "github.com/prysmaticlabs/prysm/v4/proto/eth/v2"
+	"github.com/wealdtech/go-bytesutil"
+	"go.opencensus.io/trace"
+)
+
+// GetLightClientBootstrap - implements https://github.com/ethereum/beacon-APIs/blob/263f4ed6c263c967f13279c7a9f5629b51c5fc55/apis/beacon/light_client/bootstrap.yaml
+func (bs *Server) GetLightClientBootstrap(w http.ResponseWriter, req *http.Request) {
+	// Prepare
+	ctx, span := trace.StartSpan(req.Context(), "beacon.GetLightClientBootstrap")
+	defer span.End()
+
+	// Get the block
+	blockRootParam, err := hexutil.Decode(mux.Vars(req)["block_root"])
+	if err != nil {
+		http2.HandleError(w, "invalid block root "+err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	var blockRoot [32]byte
+	copy(blockRoot[:], blockRootParam)
+	blk, err := bs.BeaconDB.Block(ctx, blockRoot)
+	if !shared.WriteBlockFetchError(w, blk, err) {
+		return
+	}
+
+	// Get the state
+	state, err := bs.Stater.StateBySlot(ctx, blk.Block().Slot())
+	if err != nil {
+		http2.HandleError(w, "could not get state "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	bootstrap, err := CreateLightClientBootstrap(ctx, state)
+	if err != nil {
+		http2.HandleError(w, "could not get light client bootstrap "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	response := &LightClientBootstrapResponse{
+		Version: ethpbv2.Version(blk.Version()).String(),
+		Data:    bootstrap,
+	}
+
+	http2.WriteJson(w, response)
+}
+
+// GetLightClientUpdatesByRange - implements https://github.com/ethereum/beacon-APIs/blob/263f4ed6c263c967f13279c7a9f5629b51c5fc55/apis/beacon/light_client/updates.yaml
+func (bs *Server) GetLightClientUpdatesByRange(w http.ResponseWriter, req *http.Request) {
+	// Prepare
+	ctx, span := trace.StartSpan(req.Context(), "beacon.GetLightClientUpdatesByRange")
+	defer span.End()
+
+	// Determine slots per period
+	config := params.BeaconConfig()
+	slotsPerPeriod := uint64(config.EpochsPerSyncCommitteePeriod) * uint64(config.SlotsPerEpoch)
+
+	// Adjust count based on configuration
+	countParam := req.URL.Query().Get("count")
+	count, err := strconv.ParseUint(countParam, 10, 64)
+	if err != nil {
+		http2.HandleError(w, fmt.Sprintf("got invalid 'count' query variable '%s', err %v", countParam, err),
+			http.StatusInternalServerError)
+		return
+	}
```

Comment:
there is a `shared.UintFromQuery` helper that you can use

Reply:
Patched.